### PR TITLE
CTCTRADERS-2345 -create page for Principal TIR Holder Id and add nav logic, update xml

### PR DESCRIPTION
**Pull request by kenneth-rayner**

### CODE REVIEW

**Review by connor-irvine — APPROVED** [review]
Test scenario written and tested locally

File: test/navigation/TraderDetailsNavigatorSpec.scala [hunk 1]
```diff
@@ -411,63 +498,52 @@ class TraderDetailsNavigatorSpec extends SpecBase with ScalaCheckPropertyChecks
 
       "must go from What is Principal's Eori page" - {
         "on a Normal Journey" - {
-          "to Check Your Answers Page if Prefix is 'GB' " in {
+          "to PrincipalName  Page if Prefix is 'GB' " in {
```
Comment (ericohwotu) [inline test/navigation/TraderDetailsNavigatorSpec.scala:501]: This looks like the same as the next test

Comment (ericohwotu) [reply test/navigation/TraderDetailsNavigatorSpec.scala:501]: Do we need one for check your answers?

File: test/forms/PrincipalTirHolderIdFormProviderSpec.scala [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    behave like mandatoryField(
+      form,
+      fieldName,
+      requiredError = FormError(fieldName, requiredKey)
+    )
+
+    behave like fieldWithInvalidCharacters(form, fieldName, invalidCharsKey, maxLength)
+
+    "must not bind strings that do not match the eoriNumber format regex" in {
```
Comment (ericohwotu) [inline test/forms/PrincipalTirHolderIdFormProviderSpec.scala:59]: should this say eoriNumber

File: app/models/domain/StringFieldRegex.scala [hunk 3]
```diff
@@ -32,4 +32,7 @@ object StringFieldRegex {
   val consignorNameRegex                   = s"^[a-zA-Z0-9&'@\\/.\\-%? ]{1,35}"
   val addressRegex: String                 = "^[a-zA-Z0-9/@?%,.\\- ]*$"
   val authorisedLocationCodeRegex          = "^[a-zA-Z0-9&'@/.%-? ]*$"
+  val principalTirHolderIdFormatRegex      = "^[a-zA-Z]{3}\\/[0-9]{3}\\/[0-9]{7}$"
+  val principalTirHolderIdCharRegex        = "^[a-zA-Z0-9]*$".r
```
Comment (ericohwotu) [inline app/models/domain/StringFieldRegex.scala:36]: is principalTirHolderIdCharRegex used?

File: test/navigation/TraderDetailsNavigatorSpec.scala (hunk 1 above)
Comment (kenneth-rayner) [reply test/navigation/TraderDetailsNavigatorSpec.scala:501]: This was in already. One is uppercase and one is lowercase,I will change the name to highlight the difference

File: app/models/domain/StringFieldRegex.scala (hunk 3 above)
Comment (kenneth-rayner) [reply app/models/domain/StringFieldRegex.scala:36]: Amended, it is now

**Review by ericohwotu — APPROVED** [review]

File: app/models/domain/StringFieldRegex.scala (hunk 3 above)
Comment (kenneth-rayner) [reply app/models/domain/StringFieldRegex.scala]: it is now

**Pull request merged**